### PR TITLE
Refactor quality score computing

### PR DESCRIPTION
**Pull request by quaxsze**
fix https://github.com/etalab/data.gouv.fr/issues/717

### CODE REVIEW

File: udata/core/dataset/models.py [hunk 1]
```diff
@@ -607,63 +607,65 @@ def quality(self):
...
+        if quality['license']:
+            score += UNIT
+        if quality['temporal_coverage']:
+            score += UNIT
+        if quality['spatial']:
+            score += UNIT
+        if quality['update_frequency']:
+            score += UNIT
+        if 'update_fulfilled_in_time' in quality:
+            if quality['update_fulfilled_in_time'] < 0:
```
Comment (maudetes) [inline udata/core/dataset/models.py:657]: `update_fulfilled_in_time` is already a boolean here

File: udata/core/dataset/models.py [hunk 2]
```diff
@@ -607,63 +607,65 @@ def quality(self):
...
-        if self.description:
-            result['description_length'] = len(self.description)
+            result['update_fulfilled_in_time'] = True if -(self.next_update - datetime.now()).days < 0 else False
+
+        result['tags_count'] = True if len(self.tags) > current_app.config.get('QUALITY_TAGS_COUNT') else False
+        result['dataset_description_quality'] = True if len(self.description) > current_app.config.get('QUALITY_DESCRIPTION_LENGTH') else False
+
         if self.resources:
             result['has_resources'] = True
             result['has_only_closed_or_no_formats'] = all(
```
Comment (maudetes) [inline udata/core/dataset/models.py:628]: Could we rename this criterion to `has_open_formats`? I find double negation confusing

File: udata/core/dataset/models.py [hunk 3]
```diff
@@ -607,63 +607,65 @@ def quality(self):
...
+        result['update_frequency'] = True if self.frequency != 'unknown' else False
         if self.next_update:
-            result['update_in'] = -(self.next_update - datetime.now()).days
-        if self.tags:
-            result['tags_count'] = len(self.tags)
-        if self.description:
-            result['description_length'] = len(self.description)
+            result['update_fulfilled_in_time'] = True if -(self.next_update - datetime.now()).days < 0 else False
+
+        result['tags_count'] = True if len(self.tags) > current_app.config.get('QUALITY_TAGS_COUNT') else False
```
Comment (maudetes) [inline udata/core/dataset/models.py:623]: Dans [le notion](https://www.notion.so/apigouv/Indice-de-score-qualit-des-jeux-de-donn-es-e226722370114093898f437808cfc551), les tags n'ont pas l'air d'être pris en compte.

File: udata/core/dataset/models.py [hunk 4]
```diff
@@ -607,63 +607,65 @@ def quality(self):
...
-                for discussion in discussions)
+                self.check_availability())  # ?
+            resource_doc = False
+            resource_desc = True
+            for resource in self.resources:
+                if resource.type == 'documentation':
+                    resource_doc = True
+                if not resource.description:
+                    resource_desc = False
+            result['resources_documentation'] = True if (resource_doc is True or resource_desc is True) else False
```
Comment (maudetes) [inline udata/core/dataset/models.py:639]:
```suggestion
            result['resources_documentation'] = resource_doc or resource_desc
```

**Review by maudetes — COMMENTED** [review]
Quality score incoming :100:

File: udata/core/dataset/models.py (hunk 3 above)
Comment (quaxsze) [reply udata/core/dataset/models.py:623]: Capture d’écran 2022-06-16 à 15 09 52

Comment (quaxsze) [reply udata/core/dataset/models.py:623]: laquelle choisit on entre les deux?

Comment (maudetes) [reply udata/core/dataset/models.py:623]: hmm indeed, je me basais sur la table, qui part du principe qu'on ignore. Ok pour toi @agarrone?

File: js/components/dataset/quality.vue [hunk 5]
```diff
@@ -27,31 +26,50 @@
...
-        :condition="quality.discussions && !quality.has_untreated_discussions"
-        :ok="_('You are currently involved in all discussions!')"
-        :ko="_('You have some untreated discussion threads. Why do not you give it a look below?')"
-        :info="_('No discussion yet? Try to involve users via social networks and/or by email.')"
-        :show-info="!quality.discussions">
-        <p>{{ _('Discussing with the community about your datasets is key to involve developers and hackers in your open process. The feedback from reusers of your data is unvaluable and should be answered rather quickly.') }}</p>
+    <qa-section :title="_('Frequency')"
+        :condition="quality.update_frequency"
+        :ok="_('That is great!')"
+        :ko="_('You currently have no frequency set for that dataset, is that pertinent?')">
```
[the remark this answers is not in the document]
Comment (maudetes) [reply js/components/dataset/quality.vue:32]: Feels like we're not helping them much, but indeed probably not worth it.

**Review by maudetes — COMMENTED** [review]
Some feedbacks, we need to make sure to be consistent between score shown and admin

File: js/components/dataset/quality.vue [hunk 6]
```diff
@@ -1,57 +1,68 @@
 <template>
 <div>
 <box :title="title" icon="thumbs-up" boxclass="box-solid" v-if="quality">
-    <doughnut :score="quality.score + 1" width="200px" height="200px"></doughnut>
```
Comment (maudetes) [inline js/components/dataset/quality.vue:4]: We can't have a full doughnut with the max score (9) today. I think we should somehow standardize the score (between 0 and 10?).

**Review by maudetes — COMMENTED** [review]
I think the last point is on the score standardization, usefull for the doughnut in admin and front gauge.

File: js/components/dataset/quality.vue (hunk 6 above)
Comment (quaxsze) [reply js/components/dataset/quality.vue:4]: WDYT in the last commit?

**Pull request merged**